File: tauri/src/schemas/explorer_event.ts
// Corresponding Backend: symbio/src/symbio_core/schemas/explorer_event.rs

export enum ExplorerEventType {
  DirChanged = 'browser/dir_changed',
  FileChanged = 'browser/file_changed',
  WatchStarted = 'watch_started',
  WatchStopped = 'watch_stopped',
  ListResult = 'list_result',
}

export interface FileChangeEvent {
  event: 'file_change';
  path: string;
  kind: string;
}

export interface WatcherErrorEvent {
  event: 'watcher_error';
  message: string;
}

export type ExplorerEvent = FileChangeEvent | WatcherErrorEvent;

export interface EventInput {
  type: ExplorerEventType;
  data: ExplorerEvent;
}

export interface StatusInput {
  type: ExplorerEventType;
  data: any;
}

File: symbio/src/symbio_core/schemas/explorer/explorer_event.rs
use serde::{Deserialize, Serialize};

/// 资源管理器事件类型
#[derive(Debug, Serialize, Deserialize)]
#[serde(tag = "event")]
pub enum Event {
    #[serde(rename = "file_change")]
    FileChange {
        path: String,
        kind: String, // "create", "modify", "remove", "rename", "other"
    },
    #[serde(rename = "watcher_error")]
    WatcherError { message: String },
}

/// 资源管理器通知类型
#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
#[serde(rename_all = "snake_case")]
pub enum ExplorerEventType {
    #[serde(rename = "browser/dir_changed")]
    DirChanged,
    #[serde(rename = "browser/file_changed")]
    FileChanged,
    #[serde(rename = "watch_started")]
    WatchStarted,
    #[serde(rename = "watch_stopped")]
    WatchStopped,
    #[serde(rename = "list_result")]
    ListResult,
}

/// 事件包裹结构（符合 Data 帧结构）
#[derive(Debug, Serialize, Deserialize)]
pub struct EventInput {
    pub r#type: ExplorerEventType,
    pub data: Event,
}

/// 生命周期事件包裹结构（用于 watch_started 等不需要 Event 数据的场景）
#[derive(Debug, Serialize, Deserialize)]
pub struct StatusInput {
    pub r#type: ExplorerEventType,
    pub data: serde_json::Value,
}
